Vagrantfile

```diff
@@ -1,71 +1,302 @@
+# $ vagrant ssh xenial64
+# vagrant@ubuntu-xenial:~$ cd ~/crystal/
+# vagrant@ubuntu-xenial:~$ make std_spec
+# ```
+#
+# The working directory is also available at /vagrant
+# but using the clone in ~/crystal helps isolating the binaries
+# from each platform and it can be faster on some circumstances
+# where files are shared with host since the clone is in the
+# native fs.
+#
+# The vagrant machines will be provisined with a native crystal
+# package or with the targz version downloaded from github.
+# Change INSTALL_GITHUB_TARGZ to choose the installation method.
+#
+# To use Makefile targets on some machines the `FLAGS=--no-debug`
+# might be needed to reduce the memory footprint.
+#
+# Notes:
+# * Running specs from /vagrant directly might cause some failures in
+#   specs related to permissions.
+# * bionic64 std_spec fails #6577
+# * xenial32 FLAGS=--no-debug is unable to run compiler_spec.
+# * trusty64 FLAGS=--no-debug is unable to build the compiler.
+# * precise32 is unable to build std_spec due to outdated libxml2. ~> 2.9.0 is required.
+# * precise32 will fail running std_spec related to reuse_port & ipv6.
+# * freebsd needs post install manual scripts in order to install crystal.
+# * alpine provisionning is failing due to openssl libressl issues.
+# * alpine is unable to to use the github .tar.gz compiler
+#
+
+INSTALL_GITHUB_TARGZ = false
+GITHUB_URL = "https://github.com/crystal-lang/crystal/releases/download/0.27.0/crystal-0.27.0-1"
+CRYSTAL_LINUX64_TARGZ = "#{GITHUB_URL}-linux-x86_64.tar.gz"
+CRYSTAL_LINUX32_TARGZ = "#{GITHUB_URL}-linux-i686.tar.gz"
+
+Vagrant.configure("2") do |config|
+  define_ubuntu config, name: 'bionic64', dist: 'bionic', bits: 64
+  define_ubuntu config, name: 'xenial64', dist: 'xenial', bits: 64
+  define_ubuntu config, name: 'xenial32', dist: 'xenial', bits: 32
+  # llvm packages > 3.8 are not available for precise/trusty. Using pre built packages
+  define_ubuntu config, name: 'trusty64', dist: 'trusty', bits: 64, llvm: { url: "http://crystal-lang.s3.amazonaws.com/llvm/llvm-3.9.1-1-linux-x86_64.tar.gz", path: "llvm-3.9.1-1" }
+  define_ubuntu config, name: 'trusty32', dist: 'trusty', bits: 32, llvm: { url: "http://crystal-lang.s3.amazonaws.com/llvm/llvm-3.9.1-1-linux-i686.tar.gz", path: "llvm-3.9.1-1" }
+  define_ubuntu config, name: 'precise64', dist: 'precise', bits: 64, llvm: { url: "http://crystal-lang.s3.amazonaws.com/llvm/llvm-3.9.1-1-linux-x86_64.tar.gz", path: "llvm-3.9.1-1" }
+  define_ubuntu config, name: 'precise32', dist: 'precise', bits: 32, llvm: { url: "http://crystal-lang.s3.amazonaws.com/llvm/llvm-3.9.1-1-linux-i686.tar.gz", path: "llvm-3.9.1-1" }
+
+  define_debian config, name: 'stretch64', dist: 'stretch', bits: 64
+  define_debian config, name: 'jessie64', dist: 'jessie', bits: 64
+
+  define_freebsd config, name: 'freebsd11', box: '11.2-STABLE'
+
+  define_alpine config, name: 'alpine64', bits: 64
+
+  config.vm.provider "virtualbox" do |vb|
+    vb.memory = 6*1024
```

Comment:
Could we make this setting configurable? Allotting 6GB is quite a lot, especially if you're running multiple VMs in parallel (or don't have much memory available anyway).
So much space is only necessary for compiling the compiler, but you can safely go with much less when working only on the stdlib.

Reply:
I left 4gb as default as before, and add env var to change the available memory